python/pyspark/sql/connect/dataframe.py

```diff
@@ -1628,11 +1629,29 @@ def checkpoint(self, *args: Any, **kwargs: Any) -> None:
     def localCheckpoint(self, *args: Any, **kwargs: Any) -> None:
         raise NotImplementedError("localCheckpoint() is not implemented.")
 
-    def to_pandas_on_spark(self, *args: Any, **kwargs: Any) -> None:
-        raise NotImplementedError("to_pandas_on_spark() is not implemented.")
-
-    def pandas_api(self, *args: Any, **kwargs: Any) -> None:
-        raise NotImplementedError("pandas_api() is not implemented.")
+    def to_pandas_on_spark(
+        self, index_col: Optional[Union[str, List[str]]] = None
+    ) -> "PandasOnSparkDataFrame":
+        warnings.warn(
+            "DataFrame.to_pandas_on_spark is deprecated. Use DataFrame.pandas_api instead.",
+            FutureWarning,
+        )
+        return self.pandas_api(index_col)
+
+    def pandas_api(
```

Comment:
nit: Shall we add a docstring for `pandas_api`?

How about
```
pandas_api.__doc__ = PySparkDataFrame.pandas_api.__doc__
```

Reply:
Nice catch. Will address the comments.